_posts/2016-03-12-make-your-data-obey.markdown

```diff
@@ -0,0 +1,200 @@
+---
+layout: post
+title: "Make Your Data Obey"
+date: 2016-03-12 10:00:00
+categories: obey data validation modelling
+author: kent.safranski
+cover: /assets/images/covers/obey.jpg
+description: Data is what programming is all about. Applications would be nothing without that beautiful i/o of data moving between modules, services, and data storage. But data is nothing if it's not consistent.
+comments: true
+disqus: true
+---
+
+Data is what programming is all about. Applications would be nothing without that beautiful i/o of data moving between modules, services, and data storage. But data is nothing if it's not consistent.
+
+**[Obey](https://github.com/TechnologyAdvice/obey) was developed specifically to address, and provide solutions for, data modelling and validation for modern web applications developed in JavaScript.**
+
+## Diving In
+
+Let's start with a simple example; validating a user account. There are some core data points we want to ensure, using Obey's `model` method to create a model object is how we start:
+
+{% highlight javascript %}
+import obey from 'obey'
+
+const user = obey.model({
+    id: { type: 'uuid', required: true },
+    email: { type: 'email', required: true },
+    password: { type: 'string', require: true }
+})
+{% endhighlight %}
+
+The above should all be fairly easy to understand. Now let's put it into action, using the `validate` method, passing the data object, then saving to a datasource. Let's assume this is going into a datasource using a CRUD library, already setup, called `db`. Using the model above:
+
+{% highlight javascript %}
+import db from './db'
+
+user.validate({
+    id: 'fe6f80c6-c530-4f10-ad0c-89f0f68a879f',
+    email: 'jdoe@email.com',
+    password: 'p@55w0rd'
+})
+// Assumes a db.create method which accepts an object as argument
+// and returns a promise
+.then(db.create).then(res => {
+    console.log('User Created!', res)
+})
+.catch(err => {
+    console.log('Failed!', err)
+})
+{% endhighlight %}
+
+Overall pretty simple stuff. We validate the model, then create the user (or catch the error). But what about that `id`? We probably want to have that generated automatically.
```

Comment:
Opportunity to highlight use of  asynchronoisily in generating Id, such as for a remote data source?